src/state/bidib_state_setter.c

```diff
@@ -52,6 +55,11 @@ void bidib_state_vendor(t_bidib_node_address node_address, uint8_t length,
 	if (mapping != NULL) {
 		t_bidib_reverser_state *reverser_state =
 				bidib_state_get_reverser_state_ref(mapping->id->str);
+		///NOTE: This is a shallow copy/not a strdup.
+		///TODO: Change to deep copy, as otherwise the lock/mutex protection is not accurate
+		// anymore: mapping is a part of the boards, whilst the reverser state is part of
+		// the trackstate.
+		// Will have to adjust all other positions where this is assigned/reset.
```

Comment:
Pleae resolve the todo

Reply:
Looks like there are some more occurrences of the same pattern for other datastructures. Will try to catch all of them.

Reply:
looks like this was actually done strategically/on purpose. Maybe better to keep the shallow copies? The state_id is not changed often, so the correct lock stuff is not too much of a problem.

Reply:
Second thought: best to adjust the allocation and freeing properly now. Some sketchy things going on with how the trackstate is freed.